quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java

```diff
@@ -267,13 +271,32 @@ void checkJdbcDriver(BuildProducer<JdbcDriverBuildItem> ignore) {
                 // We do not want to initialize the JDBC driver class
                 Class.forName(dbDriver.get(), false, Thread.currentThread().getContextClassLoader());
             } catch (ClassNotFoundException e) {
-                // Ignore queued TRACE and DEBUG messages for not initialized log handlers
-                InitialConfigurator.DELAYED_HANDLER.setBuildTimeHandlers(new Handler[]{});
-                throw new ConfigurationException(String.format("Unable to find the JDBC driver (%s). You need to install it.", dbDriver.get()));
+                throwConfigError(String.format("Unable to find the JDBC driver (%s). You need to install it.", dbDriver.get()));
             }
         }
     }
 
+    // Inspired by AgroalProcessor
+    @BuildStep
+    @Produce(CheckMultipleDatasourcesBuildStep.class)
+    void checkMultipleDatasourcesUseXA(DataSourcesBuildTimeConfig dataSourcesConfig, DataSourcesJdbcBuildTimeConfig jdbcConfig) {
+        if (dataSourcesConfig.dataSources().size() >= 2 &&
+                dataSourcesConfig.dataSources().keySet().stream()
+                        .map(ds -> jdbcConfig.dataSources().get(ds).jdbc())
+                        .filter(jdbc -> jdbc.enabled())
+                        .anyMatch(jdbc -> jdbc.transactions() != TransactionIntegration.XA)) {
```

Comment:
This should be relaxed, all but 1 source needs to be XA.

Reply:
Not sure what you mean? Which datasource doesn't need to be XA?

Reply:
Any 1 of them can be non-XA - the last commit optimization is still allowed without a warning / exception.

Reply:
Ah, thanks. Missed that. I'll update the PR as well as the [QS PR](https://github.com/keycloak/keycloak-quickstarts/pull/588) to reflect that.

Reply:
Updated the validation and release notes. The QS PR is also updated so that only the user resource is XA.